server/chat-commands/core.ts

```diff
@@ -769,10 +783,11 @@ exports.commands = {
 		}
 		battle.allowExtraction[targetUser.id].add(user.id);
 		this.addModAction(`${user.name} consents to sharing battle team and choices with ${targetUser.name}.`);
-		if (Object.keys(battle.playerTable).length === battle.allowExtraction[targetUser.id].size) {
+		if (battle.inputLog && Object.keys(battle.playerTable).length === battle.allowExtraction[targetUser.id].size) {
```

Comment:
?

Reply:
If it's possible for `battle.inputLog` to be falsy, add an error message for that case (like the others). Don't silently fail.

Reply:
checking for the log, since it can be possibly null - wasn't sure if there was a better way to handle that?

Reply:
You could also use `battle.inputLog!` to loudly fail.

But the point is, when you're writing code, think about what will happen when it gets run.

What will happen if `battle.inputLog` is falsy? No message at all, nothing to tell you what succeeded or failed. What a horrible user experience.